server/src/main/java/com/paypal/selion/grid/servlets/PasswordChangeServlet.java

```diff
@@ -57,18 +72,17 @@ private void askForCredentialsPage(PrintWriter writer) throws IOException {
     @Override
     protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
         PrintWriter writer = resp.getWriter();
-
-        String userid = (String) req.getSession().getAttribute("userId");
-        String oldPassword = req.getParameter("oldPassword");
-        String newPassword1 = req.getParameter("newPassword1");
-        String newPassword2 = req.getParameter("newPassword2");
+        // TODO Prevent userId = null. Do not allow access to this servlet unless already logged in.
```

Comment:
note the comment is partially inaccurate here.. `userId` won't ever be null.. `req.getSession()` will create a session if it does not exist. The latter part of the comment is accurate.